### PR TITLE
Move org.sonar.javascript.parser.sslr to SSLR project

### PR DESCRIPTION
**Pull request by pynicolas**
ActionParser and related classes were duplicated from sonar-java and are now factorized in SSLR: SonarSource/sslr#3
Once this is released, sonar-javascript can use that API and define only what's specific to it.

### CODE REVIEW

File: javascript-squid/src/main/java/org/sonar/javascript/parser/EcmaScriptParser.java [hunk 1]
```diff
@@ -19,24 +19,62 @@
...
+    return new EcmaScriptParser(actionParser);
+  }
+
+  @Override
+  public AstNode parse(List tokens) {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public AstNode parse(File file) {
```
Comment (SonarTech) [inline javascript-squid/src/main/java/org/sonar/javascript/parser/EcmaScriptParser.java]: MAJOR org/sonar/javascript/parser/EcmaScriptParser must not use com/sonar/sslr/api/AstNode [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AAstNode_SSLR_class_should_not_be_used)

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues:
* MAJOR 2 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [Make "ES6" a "static" method.](https://github.com/SonarCommunity/sonar-javascript/blob/6f72b2d28907c4619fc6b23a64a53e255d35cc9e/javascript-squid/src/main/java/org/sonar/javascript/parser/ActionGrammar.java#L1559) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2325)

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues:
* MAJOR 2 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [Make "ES6" a "static" method.](https://github.com/SonarCommunity/sonar-javascript/blob/6f72b2d28907c4619fc6b23a64a53e255d35cc9e/javascript-squid/src/main/java/org/sonar/javascript/parser/ActionGrammar.java#L1559) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2325)

**Comment by lindamartin** [comment]
Merged in 9488a73ef58c939c86b4d5671ff530577d932f00

**Pull request closed without merging**